coriolis/api/middleware/auth.py

```diff
@@ -14,22 +14,34 @@
 
 
 class CoriolisKeystoneContext(wsgi.Middleware):
-    @webob.dec.wsgify(RequestClass=wsgi.Request)
-    def __call__(self, req):
+    def _get_project_id(self, req):
+        if 'X_TENANT_ID' in req.headers:
+            # This is the new header since Keystone went to ID/Name
+            return req.headers['X_TENANT_ID']
+        elif 'X_TENANT' in req.headers:
+            # This is for legacy compatibility
+            return req.headers['X_TENANT']
+        return ''
```

Comment:
Instead of returning an empty string, please raise an exception if no tenant is passed, as it should be impossible to instantiate a context without a tenant:
```python
else:
    raise webob.exc.HTTPBadRequest(
        explanation=_("No 'X_TENANT_ID' or 'X_TENANT' passed."))
```